src/pages/blog/index.tsx

```diff
@@ -35,11 +35,13 @@ export const getStaticProps: GetStaticProps = async ({ preview = false }) => {
             return { frontmatter: file.frontmatter, excerpt: content, slugPath: slug.slugPath }
         })
     )
-
+    // Don't show unpublished posts in index. Proper fix coming in: https://github.com/sourcegraph/about/issues/5620
+    const publishedPosts = posts.filter(post => post.frontmatter.published)
```

Comment:
As discussed, the short term fix to hide unpublished blogposts from the index. #5620 will be the more broad bugfix.